pkg/storage/mvcc_key.go

```diff
@@ -423,3 +424,39 @@ func (k MVCCRangeKey) Validate() (err error) {
 		return nil
 	}
 }
+
+// FirstRangeKeyAbove does a binary search for the first range key at or above
+// the given timestamp. It assumes the range keys are ordered in descending
+// timestamp order, as returned by SimpleMVCCIterator.RangeKeys(). Returns false
+// if no matching range key was found.
+//
+// TODO(erikgrinaker): Consider using a new type for []MVCCRangeKeyValue as
+// returned by SimpleMVCCIterator.RangeKeys(), and add this as a method.
+func FirstRangeKeyAbove(rangeKeys []MVCCRangeKeyValue, ts hlc.Timestamp) (MVCCRangeKeyValue, bool) {
+	// This is kind of odd due to sort.Search() semantics: we do a binary search
+	// for the first range tombstone that's below the timestamp, then return the
+	// previous range tombstone if any.
+	if i := sort.Search(len(rangeKeys), func(i int) bool {
+		return rangeKeys[i].RangeKey.Timestamp.Less(ts)
+	}); i > 0 {
+		return rangeKeys[i-1], true
+	}
+	return MVCCRangeKeyValue{}, false
+}
+
+// HasRangeKeyBetween checks whether an MVCC range key exists between the two
+// given timestamps.
+func HasRangeKeyBetween(rangeKeys []MVCCRangeKeyValue, upper, lower hlc.Timestamp) bool {
+	if len(rangeKeys) == 0 {
+		return false
+	}
+	if upper.Less(lower) {
```

Comment:
Yeah, good point. I was thinking that given the literal meaning of `HasRangeKeyBetween()`, it doesn't really matter which order the timestamps are in, but it very much does for e.g. range tombstones. I've changed it now so that it simply returns `false` if they're in the wrong order, and added a panic assertion when running under race.